### PR TITLE
Credential-store commands

### PR DESCRIPTION
**Pull request by jtymel**
Please do not merge yet. At least we need to make clear what is the reason behind not being able to use CamelCase credential store aliases (see comment in AddCredentialStoreAliasOnlineTest)

### CODE REVIEW

File: commands/src/main/java/org/wildfly/extras/creaper/commands/elytron/credentialstore/AddCredentialStoreAlias.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    public static enum EntryType {
+
+        OTHER("Other"), PASSWORD_CREDENTIAL("org.wildfly.security.credential.PasswordCredential");
+
+        private final String entryTypeName;
+
+        private EntryType(String versionComparisonName) {
+            this.entryTypeName = versionComparisonName;
```
Comment (hsvabek) [inline commands/src/main/java/org/wildfly/extras/creaper/commands/elytron/credentialstore/AddCredentialStoreAlias.java]: change name of argument

File: testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/elytron/credentialstore/AddCredentialStoreAliasOnlineTest.java [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+@RunWith(Arquillian.class)
+public class AddCredentialStoreAliasOnlineTest extends AbstractElytronOnlineTest {
+
+    private static final String TEST_CREDENTIAL_STORE_NAME = "CreaperTestCredentialStore";
+    private static final Address TEST_CREDENTIAL_STORE_ADDRESS = SUBSYSTEM_ADDRESS
+            .and("credential-store", TEST_CREDENTIAL_STORE_NAME);
+
+    //TODO: cannot be CamelCase, why??
+    private static final String TEST_CREDENTIAL_STORE_ALIAS_NAME = "creapertestcredentialstorealias";
```
Comment (hsvabek) [inline testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/elytron/credentialstore/AddCredentialStoreAliasOnlineTest.java:24]: CS support only lowercase entry and aliases are converted to lowercase. If you use CLI manually it works fine.
But in tests is some kind of problem with assertion -> comparion original address with converted to lowercase -> assertion FALSE.
I have it on TODO list.

File: testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/elytron/credentialstore/AddCredentialStoreAliasOnlineTest.java [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    }
+
+    @Ignore("https://issues.jboss.org/browse/JBEAP-6611")
+    @Test
+    public void addFullCredentialStoreAlias() throws Exception {
+        AddCredentialStoreAlias addCredentialStoreAlias
+                = new AddCredentialStoreAlias.Builder(TEST_CREDENTIAL_STORE_ALIAS_NAME)
+                .credentialStore(TEST_CREDENTIAL_STORE_NAME)
+                .secretValue("someSecretValue")
+                .entryType(AddCredentialStoreAlias.EntryType.OTHER)
```
Comment (hsvabek) [inline testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/elytron/credentialstore/AddCredentialStoreAliasOnlineTest.java]: default value for Entry type is PasswordCredential

File: testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/elytron/credentialstore/AddCredentialStoreAliasOnlineTest.java (hunk 2 above)
Comment (hsvabek) [inline testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/elytron/credentialstore/AddCredentialStoreAliasOnlineTest.java:24]: It's on my TODO list.
Explanation of this is in our etherpad about CS.

**Comment by jtymel** [comment]
@hsvabek does the last commit fix all issues you have found? If so, I'll squash last two commits and merge

**Comment by hsvabek** [comment]
It's ok. You can merge it. Thanks.

**Pull request merged**